www/docs/en/dev/plugin_ref/spec.md

```diff
@@ -17,700 +17,489 @@ license: >
+  Example:
+  ```
+  <js-module src="socket.js" name="Socket">
+    <clobbers target="chrome.socket" />
+  </js-module>
+  ```
+  Here module.exports gets inserted into the `window` object as `window.chrome.socket`.
 
-## _license_ Element
+#### merges
 
-Plugin license. The text content of the element contains the plugin license. An example:
+  Allowed within `<js-module>` element. Used to specify the namespace under `window` object where module.exports gets merged with any existing value. If any key already exists, the module's version overrides the original. You can have as many `<merges>` as you like. Any object not available on `window` is created.
 
-    <license>Apache 2.0 License</license>
+  Attributes(type) | Description
+  ---------------- | ------------
+  target(string) | The namespace which module.exports gets merged to.
 
-## _asset_ Element
+  Example:
+  ```
+  <js-module src="socket.js" name="Socket">
+    <merges target="chrome.socket" />
+  </js-module>
+  ```
+  Here module.exports gets merged with any existing value at `window.chrome.socket`.
 
-One or more elements listing the files or directories to be copied
-into a Cordova app's `www` directory. Examples:
+#### runs
 
-    <!-- a single file, to be copied in the root directory -->
-    <asset src="www/foo.js" target="foo.js" />
-    <!-- a directory, also to be copied in the root directory -->
-    <asset src="www/foo" target="foo" />
+  Allowed within `<js-module>` element. It implies that your code should be specified with `cordova.require`, but not installed on the `window` object. This is useful when initializing the module, attaching event handlers or otherwise. You can only have up to one `<runs/>` tag. Note that including a `<runs/>` with `<clobbers/>` or `<merges/>` is redundant, since they also `cordova.require` your module.
 
-All `<asset>` tags require both `src` and `target` attributes.
-Web-only plugins contains mostly `<asset>` elements. Any `<asset>`
-elements that are nested within `<platform>` elements specify
-platform-specific web assets, as described below. Attributes include:
+  Example:
+  ```
+  <js-module src="socket.js" name="Socket">
+    <runs/>
+  </js-module>
+  ```
 
-* `src` (required): 
-  Where the file or directory is located in the plugin package,
-  relative to the `plugin.xml` document.  If a file does not exist at
-  the specified `src` location, plugman stops and reverses the
-  installation process, issues a notification about the conflict, and
-  exits with a non-zero code.
+### dependency
 
-* `target` (required):
+  The `<dependency>` tag allows you to specify other plugins on which the current plugin depends. While future versions will access them from plugin repositories, in the short term plugins are directly referenced as URLs by `<dependency>` tags. In the future, version constraints will be introduced, and a plugin repository will exist to support fetching by name instead of explicit URLs.
```

Comment:
I think we support fetching by name (see https://github.com/apache/cordova-plugin-file-transfer/blob/master/plugin.xml#L32). Also, the bit about reverse domain style is outdated